module/zcommon/zfs_comutil.c

```diff
@@ -207,10 +208,48 @@ const char *zfs_history_event_names[ZFS_NUM_LEGACY_HISTORY_EVENTS] = {
 	"pool split",
 };
 
+#if defined(_KERNEL)
+#define	gettext(x) x
+#endif
```

Comment:
We should avoid allowing the localization logic to seep down in to the shared user/kernel code.  Which means in this case we're going to have to live with a little code duplication.  I'd suggest moving the `zpool_state_to_name` back to where it was and adding a new `spa_state_to_name(spa_t *)` function without `gettext` to `spa_misc.c`.  This has the additional advantage of not disrupting the existing `libzfs` API.